streams/tests/test_core.py

```diff
@@ -494,3 +494,33 @@ def test_partition_str():
 def test_stream_name_str():
     source = Stream(name='this is not a stream')
     assert str(source) == '<this is not a stream; Stream>'
+
+
+def test_piping():
+    source = Stream()
+    source2 = Stream()
+    # put source2 upstream of source
+    source2.parents = [source]
+
+    L = list()
+    source.map(L.append)
+
+    source2.emit(1)
+    source2.emit(3)
+
+    assert L == [1,3]
```

Comment:
you're right, I'll remove it (it's what `connect` is effectively doing behind the scenes).